twenty-first/src/shared_math/rescue_prime_regular.rs

```diff
@@ -1038,6 +1038,7 @@ impl AlgebraicHasher for RescuePrimeRegular {
 }
 
 impl SpongeHasher for RescuePrimeRegular {
+    const RATE: usize = RATE;
```

Comment:
Adding `RATE` to the trait seems fine, but references to `RATE` in the functions of this trait need to be `Self::RATE` unless they should refer to one of the other `RATE`s that exist (each hash function has its own `pub const RATE: usize`, and there is a shared one in algebraic_hasher.rs.